docs/multiregion.md

```diff
@@ -0,0 +1,98 @@
+
+Between regions the following things should be synced, and should be globally unique:
+* `(ServiceCluster, Account)`
+
+Additionally the following objects should be replicated across every region:
+* `(Account)`
+
+Thus there must exist global consensus what the global KubeCarrier configuration should be. The authorization shall be handled the same as in the single region case --> management cluster shall create the necessary roles and rolebinding for the account.
+
+This is enough to handle the authorization via the native k8s mechanism. For Authentication see the dedicated section.
+
+## related work:
+
+* https://github.com/kubermatic/kubecarrier/pull/291/files
+
+This solutions revolves around having a single superCluster managing multiple regions with, and each region being managed by region-local installation.
+
+Pros:
+* the user authentication happens at the superCluster level, whilst their actions is propagated to region-local management cluster via the `Impersonate` headers
+* installation is handled via the superCluster operator
+
+Cons:
+* the latency issue still remains. (e.g. request for creating new service instance would propagate from user -- superClusterManager -- region -- region local serviceCluster)
+* the superCluster becomes Single point of Failure
+* the superCluster has admin privileges in the region's management cluster. A exploit in our API server implementation could lead to unwanted privilege escalation.
+
+## Proposed solution:
+
+### Having no superCluster/Federations
+
+* The latency and SPOF issue shall still reamain.
+* We build API server accepting user's kubeconfig/token
+    * This way we're preventing privilege escalation since API server doesen't have cluster-admin privileges.
+* it's simpler approach
+
+### build truly decentralized KubeCarrier installation with helper API/library for cross-region management
+
+The way cross-region requires consists of two operation types:
+* routing
+* fan-out
+
+Also there are two ways for keeping configuration in sync:
+* static configuration
+* central configuration plane (akin to superCluster in the original @nico proposal)
+
+The routing operations require the client be routed to the right region executing its request. (e.g. create new serviceCluster, tenant consuming new service, etc.). The typical workflow would work like this:
+
+* client looking up the cache does he have a route in place
+    * if not ask the central configuration plane
+    * alternatively fail since the region are hard-coded for the client
+* handle the authentication. The authorization is already handled by the account, and general KubeCarrier installation. The authentication could work in multiple ways:
+    * each kubecarrier region management cluster having the same credentials (e.g. same OIDC settings, or same client CA, or whatever)
+* perform the required operation
+
+The fan-out operations are more complicated, but rarer. They include the following:
```

Comment:
I agree with you that adding/removing regions/accounts are rare operations, but why do we introduce extra complexities just for handling the rare operations?

Reply:
This is a bit brain-storming. I'm more in favor of having a central configuration plane for regions/accounts that these semi-rare complex operations are handled properly without distributed shenanigans and everything else as distributed as possible.